web/app/js/components/NavigationResources.jsx

```diff
@@ -19,18 +19,35 @@ const styles = () => ({
   navMenuItem: {
     paddingLeft: "24px",
     paddingRight: "24px",
-  },
+  }
 });
 
 class NavigationResourcesBase extends React.Component {
   static propTypes = {
     classes: PropTypes.shape({}).isRequired,
-    data: PropTypes.arrayOf(metricsPropType.isRequired).isRequired,
-  }
+    data: PropTypes.arrayOf(metricsPropType.isRequired).isRequired
+  };
 
   constructor(props) {
     super(props);
-    this.state = {open: false};
+    this.state = {
+      open: false,
+      allMetrics: {},
+      nsMetrics: {}
+    };
+  }
+
+  componentWillReceiveProps(prevProps, nextProps) {
```

Comment:
The changes you made in NavigationResource.jsx were enough to fix the bug, and moving the setting of this state into a `componentWillReceiveProps` isn't necessary.  Reverting this to the way it was, I still see new resources being added and removed in the sidebar.

(Also note that `componentWillReceiveProps` will be deprecated in the future, so we should try to avoid introducing them. https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops)

Reply:
Yeah, I shouldn't have done that because data is already coming as props, so every time props change component will re-render.